pgmax/factors/enumeration.py

```diff
@@ -179,12 +180,13 @@ def concatenate_wirings(wirings: Sequence[EnumerationWiring]) -> EnumerationWiri
 
 def compile_enumeration_wiring(
     factor_edges_num_states: np.ndarray,
-    variables_for_factors: Tuple[Tuple[nodes.Variable, ...], ...],
+    variables_for_factors: Tuple[nodes.Variable, ...],
```

Comment:
I previously changed this because using a tuple of tuples can help us get rid of several unnecessary attributes (e.g. `factor_sizes` which I got rid of, and `num_factors` which I forgot to get rid of. Unless making it a tuple of tuples somehow makes it slow, I think having as close to a minimal set of nonredundant arguments is better.

Reply:
making it a tuple of tuple makes it slower, which is why I have changed it